src/engine_api/execution_payload.zig

```diff
@@ -1,13 +1,104 @@
 const std = @import("std");
 const types = @import("../types/types.zig");
 const lib = @import("../lib.zig");
+const version = @import("../version.zig");
 const Blockchain = lib.blockchain.Blockchain;
 const state = lib.state;
 const Allocator = std.mem.Allocator;
 const BlockHeader = types.BlockHeader;
 const Withdrawal = types.Withdrawal;
 const Tx = types.Tx;
 
+pub const PayloadAttributes = struct {
+    timestamp: u64,
+    random: types.Hash32,
+    suggested_fee_recipient: types.Hash32,
+    withdrawals: []*types.Withdrawal,
+    beacon_root: ?types.Hash32,
+};
+
+pub const StatelessPayloadStatusV1 = struct {
+    status: []const u8,
+    state_root: types.Hash32,
+    receipt_root: types.Hash32,
+    validator_error: ?[]const u8,
+};
+
+pub const ExecutionPayloadEnvelope = struct {
+    execution_payload: *ExecutionPayload,
+    block_value: *types.Hash32,
+    blobs_bundle: *BlobsBundleV1,
+    Requests: [][]const u8,
+    override: bool,
+    witness: []const u8,
+};
+
+pub const BlobAndProofV1 = struct {
+    blob: []const u8,
+    proof: []const u8,
+};
+
+pub const BlobsBundleV1 = struct {
+    commitments: []types.Hash32,
+    proofs: []types.Hash32,
+    blobs: []types.Hash32,
+};
+
+pub const PayloadStatusV1 = struct {
+    status: []const u8,
+    witness: []const u8,
+    latest_valid_hash: ?types.Hash32,
+    validattion_error: ?[]const u8,
+};
+
+pub const TransitionConfigurationV1 = struct {
+    terminal_total_difficulty: []const u8,
+    terminal_block_hash: types.Hash32,
+    terminal_block_number: u64,
+};
+
+const PayloadVersion = enum(u8) {
+    PayloadV1 = 1,
+    PayloadV2,
+    PayloadV3,
+};
+
+pub const PayloadID = struct {
+    const Self = @This();
+    inner: [8]u8,
+
+    pub fn version(pid: *Self) PayloadVersion {
+        return @as(PayloadVersion, @enumFromInt(pid.inner[0]));
+    }
+
+    pub fn string(pid: *Self, out: []u8) !void {
+        if (out.len != 16) {
+            return error.OutputBufferOverflow;
```

Comment:
Nit: error name sounds weird since not necessarily is an overflow. Maybe `OutputBufferInvalidSize`?